interfaces/repo.go

```diff
@@ -46,14 +45,18 @@ type Repository struct {
 
 // NewRepository creates an empty plug repository.
 func NewRepository() *Repository {
-	return &Repository{
+	repo := &Repository{
 		ifaces:    make(map[string]Interface),
 		plugs:     make(map[string]map[string]*Plug),
 		slots:     make(map[string]map[string]*Slot),
 		slotPlugs: make(map[*Slot]map[*Plug]*Connection),
 		plugSlots: make(map[*Plug]map[*Slot]*Connection),
 		backends:  make(map[SecuritySystem]SecurityBackend),
 	}
+	snap.SanitizePlugsSlots = func(snapInfo *snap.Info) error {
```

Comment:
Could you please add a check that `snap.SanitizePlugsSlots` is not set yet? I worry that we may end up with sanitizer that checks against some ghost repository for *whatever* reason later.

Reply:
I tried a couple of different approches revolving around such check, but it turned out to be impossible due to the fact that in some tests we create two instances of repo side-by-side, or two daemons (with repos created implicitely).
Anyway, I changed this to an explicit setup call from ifacemgr, that way we know we do this only once in real code and there is no possiblity to mess it up. And ReadInfo will panic if the callback is not set.

Reply:
Thank you